pkg/tls/certRenewer.go

```diff
@@ -372,46 +321,35 @@ func (c *CertRenewer) ValidCert() (bool, error) {
 	return true, nil
 }
 
-// IsKyvernoInRollingUpdate returns true if Kyverno is in rolling update
-func IsKyvernoInRollingUpdate(deploy *appsv1.Deployment, logger logr.Logger) bool {
-	var replicas int32 = 1
-	if deploy.Spec.Replicas != nil {
-		replicas = *deploy.Spec.Replicas
-	}
-	nonTerminatedReplicas := deploy.Status.Replicas
-
-	if nonTerminatedReplicas > replicas {
-		logger.Info("detect Kyverno is in rolling update, won't trigger the update again")
-		return true
+// RollingUpdate triggers a rolling update of Kyverno pod.
+// It is used when the rootCA is renewed, the restart of
+// Kyverno pod will register webhook server with new cert
+func (c *CertRenewer) RollingUpdate() error {
```

Comment:
Do we need this anymore? Since now webhook server dynamically fetches the TLS key pair?

Reply:
I think we still need it.
I would like to extend validation to user provided certs too (this should not be restricted to kyverno managed ones)

Reply:
Ok.